server/src/queries.rs

```diff
@@ -15,10 +17,19 @@ pub async fn get_signals(
     end: i64,
     network: crate::types::Network,
 ) -> anyhow::Result<Vec<Signal>> {
+    let allowed_origins = env::var("ALLOWED_ORIGINS")
+        .expect("ALLOWED_ORIGINS not set");
+    let allowed_origins: Vec<&str> = allowed_origins
+        .split(",")
+        .map(str::trim)
+        .collect();
+
     if let Some(cached) = cache.get(&format!("{}-{}-{}", start, end, network)) {
         Ok(cached.to_vec())
     } else {
-        let signals = sqlx::query_as!(
+        let signals = vec![];
+        for _origin in allowed_origins {
```

Comment:
You don't need to iterate over the allowed origins. The allowed origins is a HTTP construct and is basically a gatekeeper on what domains are allowed to hit this endpoint. The old functionality was correct.